### PR TITLE
major: #200 Shared nothing architecture - Partition Events

### PR DESCRIPTION
**Pull request by astubbs**
** Instead of shared state being manipulated directly (assigned partitions, and removal of dangling work), it's now
down through the Actor model
** This simplifies the system, removes potential for errors with shared state and race conditions

Check:
- [ ] BrokerPoller can't access state - when performing commit, must request or be sent commit offset data by state owner (#419 )
- [ ] ShardManager#totalShardEntriesNotInFlight

Implements:
- #200 
- Removes onPartitionsLost handlers as we don't need to do anything differently than from onPartitionsRevoked

Related:
- #271 

Blocked by:
- #325

Causes:
- #419

### CODE REVIEW

**Review by rkolesnev — COMMENTED** [review]
Looks good to me semi-understanding the bigger picture - looks like structure change with no logic change :) 
ControllerEventMessage is composite of different events - eventually would be nice to split into event message per type of event.
Similarly PartitionEvent part ControllerEventMessage is composite of Assigned and Revoked actions.

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 1]
```diff
@@ -100,31 +101,42 @@
...
+
     /**
      * An inbound message to the controller.
      * <p>
      * Currently, an Either type class, representing either newly polled records to ingest, or a work result.
      */
     @Value
     @RequiredArgsConstructor(access = PRIVATE)
-    private static class ControllerEventMessage<K, V> {
+    protected static class ControllerEventMessage<K, V> {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:113]: extract

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 2]
```diff
@@ -100,31 +101,42 @@
...
         EpochAndRecordsMap<K, V> consumerRecords;
+        ConsumerRebalanceHandler.PartitionEventMessage partitionEventMessage;
 
         private boolean isWorkResult() {
             return workContainer != null;
         }
 
         private boolean isNewConsumerRecords() {
-            return !isWorkResult();
+            return consumerRecords != null;
```
[the remark this answers is not in the document]
Comment (astubbs) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java]: removed

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java (hunk 1 above)
Comment (astubbs) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java]: removed

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 3]
```diff
@@ -265,36 +269,39 @@ private void checkNotSubscribed(org.apache.kafka.clients.consumer.Consumer<K, V>
...
     @Override
     public void subscribe(Pattern pattern, ConsumerRebalanceListener callback) {
         log.debug("Subscribing to {}", pattern);
         usersConsumerRebalanceListener = Optional.of(callback);
-        consumer.subscribe(pattern, this);
+        consumer.subscribe(pattern, this.rebalanceHandler);
+    }
+
+    public void onPartitionsRevokedTellAsync(Collection<TopicPartition> partitions) {
+        getMyActor().tell(controller -> controller.onPartitionsRevokedInternal(partitions));
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:296]: tell or tell immediately?

**Comment by eddyv** [comment]
Closing - Stale.

**Pull request closed without merging**